File: be/pnt_spring/src/main/java/com/pnt/pnt_spring/domain/games/game/api/resp/GameResultResponse.java
package com.pnt.pnt_spring.domain.games.game.api.resp;

import java.time.OffsetDateTime;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class GameResultResponse {
	private Long gameId;
	private String winner;
	private OffsetDateTime endedAt;
	private TotalStats stats;
	private MvpResponse mvp;
	private MvpResponse winningSecond;
	private MvpResponse losingFirst;

	private GameMemberStat myStat;

	@Getter
	@Builder
	public static class TotalStats {
		private int arrests;
		private int missionsCleared;
		private int durationSec;
	}

	@Getter
	@Builder
	public static class MvpResponse {
		private Long memberId;
		private String nickname;
		private String role;
		private String description;

		private Integer walk;
		private Integer arrestCount;
		private Integer longestSurvived;

		private String rank;
		private Integer maxArrestCount;
		private Integer maxSurvivalTime;
	}

	@Getter
	@Builder
	public static class GameMemberStat {
		private Long memberId;
		private String nickname;
		private String role;

		// 이번 게임 기록
		private Integer walk;
		private Integer arrestCount;     // 이번 판 체포 수
		private Integer longestSurvived; // 이번 판 생존 시간

		private String rank;             // 현재 등급 (예: "브론즈", "실버")
		private Integer maxArrestCount;  // (경찰일 때) 개인 최고 체포 기록
		private Integer maxSurvivalTime; // (도둑일 때) 개인 최고 생존 시간
	}
}

File: fe/D104/app/src/main/java/com/d104/pnt/data/remote/model/response/GameResultResponse.kt
package com.d104.pnt.data.remote.model.response

data class GameResultResponse(
    val gameId: Long,
    val winner: String,
    val endedAt: String,
    val stats: GameStats,
    val mvp: PlayerResult?,
    val winningSecond: PlayerResult?,
    val losingFirst: PlayerResult?,
    val myStat: GameResultMyStat
)

data class GameStats(
    val arrests: Int,
    val missionsCleared: Int,
    val durationSec: Long
)

data class PlayerResult(
    val memberId: Long,
    val nickname: String,
    val role: String,
    val description: String
)

data class GameResultMyStat(
    val memberId: Long,
    val nickname: String,
    val role: String,
    val walk: Int,
    val arrestCount: Int,
    val longestSurvived: Int,
    val rank: String,
    val maxArrestCount: Int,
    val maxSurvivalTime: Int
)